front/app/modules/commercial/customizable_homepage_banner/admin/CallToActionButtonSettings.tsx

```diff
@@ -14,39 +14,62 @@ import {
 } from 'services/appConfiguration';
 
 interface SigninStateProps {
-  options: string[];
+  options: (CTASignedOutButton | CTASignedInButton)[];
   currentValue: string;
   signInState: 'not_signed_in' | 'signed_in';
+  labelMessage: MessageDescriptor;
 }
 
 const CallToActionSettingsSigninState = ({
   options,
   currentValue,
   signInState,
+  labelMessage,
 }: SigninStateProps) => (
   <SectionField>
     <Label>
-      <FormattedMessage
-        {...{
-          signed_in: messages.signed_in,
-          not_signed_in: messages.not_signed_in,
-        }[signInState]}
-      />
+      <FormattedMessage {...labelMessage} />
     </Label>
     {options.map((option) => (
       <Radio
         key={option}
         onChange={() => {}}
         currentValue={currentValue}
         value={option}
-        label={<FormattedMessage {...messages[option]} />}
+        label={
+          <FormattedMessage
+            {...{
+              sign_up_button: messages.sign_up_button,
+              customized_button: messages.customized_button,
+              no_button: messages.no_button,
+            }[option]}
+          />
+        }
         name={`call_to_action_${signInState}_selected_option`}
-        id={`locale-${currentValue}`}
+        id={`${signInState}-${currentValue}`}
       />
     ))}
   </SectionField>
 );
 
+const CTASettingsSignedIn = ({ currentValue }) => (
+  <CallToActionSettingsSigninState
+    options={CALL_TO_ACTION_SIGNED_IN_OPTIONS}
+    currentValue={currentValue}
+    signInState="signed_in"
+    labelMessage={messages.signed_in}
+  />
+);
+
+const CTASettingsSignedOut = ({ currentValue }) => (
+  <CallToActionSettingsSigninState
+    options={CALL_TO_ACTION_NOT_SIGNED_IN_OPTIONS}
```

Comment:
I would also change this one to SIGNED_OUT then?

Reply:
Sure, will change it in all the places.